core/src/com/unciv/logic/automation/SpecificUnitAutomation.kt

```diff
@@ -280,6 +287,79 @@ object SpecificUnitAutomation {
         }
     }
 
+    fun automateMissionary(unit: MapUnit){
+        if (unit.religion != unit.civInfo.religionManager.religion?.name)
+            return unit.destroy()
+
+        val cities = unit.civInfo.gameInfo.getCities().asSequence()
+            .filter { it.religion.getMajorityReligion()?.name != unit.getReligionDisplayName() }
+            .filterNot { it.civInfo.isAtWarWith(unit.civInfo) }
+            .minByOrNull { it.getCenterTile().aerialDistanceTo(unit.currentTile) } ?: return
+
+
+        val destination = cities.getTiles().asSequence()
+            .filterNot { unit.getTile().owningCity == it.owningCity } // to prevent the ai from moving around randomly
+            .filter { unit.movement.canMoveTo(it) }
+            .minByOrNull { it.aerialDistanceTo(unit.currentTile) }
+
+
+        if (destination != null) {
+            unit.movement.headTowards(destination)
+        }
+
+        if (unit.currentTile.owningCity?.religion?.getMajorityReligion()?.name != unit.religion)
+            doReligiousAction(unit, unit.getTile())
+    }
+
+    fun automateInquisitor(unit: MapUnit){
+        val cityToConvert = unit.civInfo.cities.asSequence()
+            .filterNot { it.religion.getMajorityReligion()?.name == null }
+            .filterNot { it.religion.getMajorityReligion()?.name == unit.religion }
+            .minByOrNull { it.getCenterTile().aerialDistanceTo(unit.currentTile) }
+
+        val cityToProtect = unit.civInfo.cities.asSequence()
+            .filter { it.religion.getMajorityReligion()?.name == unit.religion }
+            .filter { isInquisitorInTheCity(it, unit)}
+            .maxByOrNull { it.population.population }  // cities with most populations will be prioritized by the AI
+
+        var destination: TileInfo? = null
+
+        if (cityToProtect != null){
+            destination = cityToProtect.getCenterTile().neighbors.asSequence()
+                    .filterNot { unit.getTile().owningCity == it.owningCity } // to prevent the ai from moving around randomly
+                    .filter { unit.movement.canMoveTo(it) }
+                    .minByOrNull { it.aerialDistanceTo(unit.currentTile) }
+        }
+        else if (cityToConvert != null){
```

Comment:
How about instead of "else if", we do "if destination is not null"? City to protect may have no movable tiles

Reply:
sure fixed